### PR TITLE
fix: migrate CI, AWS account and alerting

### PR DESCRIPTION
**Pull request by xav**
# Description

- Migrate to the dedicated AWS account
- Replace the CI with the standard one
- Update app configuration to avoid "hidden" configuration values.
- Add CloudWatch alerts routed to BetterStack

The PR looks insanely huge, but a lot of the rust changes are cosmetic, due to switching `cargofmt` to the stable channel. The actual changes are around configuration management and geoip resolver instantiation.

Regarding the Terraform code and GitHub actions, they've basically been replaced so it might be easier to pull the branch and look at the code instead of trying to check the diffs.

The configured hosted zone is still the `wl-blockchain` temporary one to validate the migration and must be updated later.

## How Has This Been Tested?

Deployed manually.

The logs show 404 errors when trying to update the weights from Prometheus, but I assume that because there's no traffic, the instance is still empty.

### CODE REVIEW

**Review by arein — COMMENTED** [review]
Do we need to announce maintenance for this?

**Comment by xav** [comment]
> Do we need to announce maintenance for this?

Not right now, this will deploy to `wl-blockchain.walletconnect.com` and leave `rpc.walletconnect.com`.
Once it'ss validated, we can switch the DNS. That's when we'll need to make an announcement

**Review by geekbrother — APPROVED** [review]
lgtm

File: .github/workflows/dispatch_validate.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      stage-url: https://staging.${{ vars.SUBDOMAIN_NAME }}.walletconnect.com/health
+
+  validate-prod:
+    name: Validate - Prod
+    if: ${{ inputs.check-prod }}
+    uses: ./.github/workflows/sub-validate.yml
+    secrets: inherit
+    with:
+      stage: prod
+      stage-url: https://staging.${{ vars.SUBDOMAIN_NAME }}.walletconnect.com/health
```
Comment (chris13524) [inline .github/workflows/dispatch_validate.yaml:56]:
```suggestion
      stage-url: https://${{ vars.SUBDOMAIN_NAME }}.walletconnect.com/health
```

File: rustfmt.toml [hunk 2]
```diff
@@ -1,18 +1,22 @@
...
-group_imports = "StdExternalCrate"
 use_try_shorthand = true
-wrap_comments = true
-overflow_delimited_expr = true
 remove_nested_parens = true
 reorder_modules = true
-unstable_features = true
 use_field_init_shorthand = true
+
+## We only use settings available in the stable channel
```
Comment (chris13524) [inline rustfmt.toml:9]: Darn I really liked these format options, especially the ones related to imports.

File: .github/workflows/sub-app-check.yml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Run sccache-cache
+        uses: mozilla-actions/sccache-action@v0.0.3
+
+      - name: "Clippy"
+        uses: WalletConnect/actions-rs/cargo@1.0.0
+        with:
+          command: clippy
+          args: -- -D warnings
```
Comment (chris13524) [inline .github/workflows/sub-app-check.yml:41]:
```suggestion
          args: --all-features --tests -- -D clippy::all
```

**Pull request closed without merging**